client/request/handlers.go

```diff
@@ -0,0 +1,286 @@
+package request
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"regexp"
+	"strconv"
+
+	httpsignatures "github.com/99designs/httpsignatures-go"
+	dd_opentracing "github.com/DataDog/dd-trace-go/opentracing"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+	"github.com/remind101/pkg/httpx"
+)
+
+// Handlers represents lists of request handlers for each phase in the request
```

Comment:
It's be great to have a handler_test, mostly as an example of how to write tests for new handlers.

Reply:
Nevermind. found them in request_test.

Though the fact I couldn't find them at first might be a signal on discoverability? Or I'm just not a great reader. 😄

Reply:
No that's a good point, I'll move them :) Perhaps could write them in a more isolated way, but I'll just move them for now